cmd/changelog-parser/main.go

```diff
@@ -225,6 +238,8 @@ func runParser(options cliOptions) {
 		repoConfig.SelectUnreleased()
 	}
 
+	options.generateOutputFilename()
```

Comment:
Probably not relevant since this struct will probably move out into its own package but we generally don't use this self-modifying invocation style as we can either have field accessors and/or calculate the resulting values in constructors (repositories config is a special case since we derive it from YAML if that is where you are replicating this pattern from).